source/darwin/darwin_pki_utils.c

```diff
@@ -45,7 +46,34 @@ int aws_import_public_and_private_keys_to_identity(
 
     SecCertificateRef certificate_ref = NULL;
     SecKeychainRef import_keychain = NULL;
-    SecKeychainCopyDefault(&import_keychain);
+
+    if (keychain_path) {
+        OSStatus keychain_status = SecKeychainOpen(keychain_path, &import_keychain);
+        if (keychain_status != errSecSuccess) {
+            AWS_LOGF_ERROR(
+                AWS_LS_IO_PKI,
+                "static: error opening keychain \"%s\" with OSStatus %d",
+                keychain_path,
+                keychain_status);
+            return AWS_OP_ERR;
+        }
+        keychain_status = SecKeychainUnlock(import_keychain, 0, "", TRUE);
```

Comment:
can we use c99 standard types here? as in pass the stdint.h's bool (true)

Reply:
Sure, I'll change it.